File: Chapter9_Example/src/main/java/UserFactory.java
public class UserFactory {
    public static User create(Object[] data) {
        assert data.length >= 3;

        int id = (int)data[0];
        String email = (String)data[1];
        UserType type = (UserType)data[2];

        return new User(id, email, type);
    }
}

File: unit-testing/src/main/kotlin/book/chapter10/unitofwork/UserFactory.kt
package book.chapter10.unitofwork

class UserFactory {

    companion object {
        fun create(data: Array<Any>): User {
            require(data.size >= 3)

            val id = data[0] as Int
            val email = data[1] as String
            val type = data[2] as UserType

            return User(id, email, type, false)
        }
    }
}